trollflow2/plugins/__init__.py

```diff
@@ -216,6 +217,19 @@ def prepared_filename(fmat, renames):
         yield orig_filename
 
 
+def format_decoration(fmat, fmat_config):
+    """Format decoration text using template given in fmt_config with key-value pairs in fmat."""
+    fmat_config_local = copy.deepcopy(fmat_config)
+    if "decorate" in fmat_config:
+        for deco in fmat_config_local["decorate"]["decorate"]:
+            if "text" in deco and "txt" in deco["text"]:
+                try:
+                    deco["text"]["txt"] = deco["text"]["txt"].format(**fmat)
+                except KeyError:
+                    LOG.warning('Could not format: %s.', str(deco["text"]["txt"]))
```

Comment:
This could be a separate (private) function `_format_decoration_text(deco, fmat)`. It would make the code a bit cleaner.